File: Substring Counter/SubstringCounter.py
sub_string = 'bob'
sub_string_length = len(sub_string)

s <- "jhfuhiuahiahobobob"

bobs = 0

for i in range(len(s)):
    if s[i : i + sub_string_length] == sub_string:
        bobs += 1
    else:
        next
print("Number of times bob occurs is: " + str(bobs))

File: Substring Counter/SubstringCounter.java
public class SubstringCounter {

   public static void main(String[] args) {

     String s = "jhfuhiuahiahobobob";
     String subString = "bob";

     int lastIndex = 0;
     int count = 0;

     while( lastIndex != -1 ) {

         lastIndex = s.indexOf(subString, lastIndex);

         if( lastIndex != -1 ) {
             count ++;
             lastIndex += subString.length();
         }
     }
     System.out.println(count);
   }
}
